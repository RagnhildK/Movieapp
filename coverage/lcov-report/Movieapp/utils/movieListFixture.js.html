
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for Movieapp/utils/movieListFixture.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> / <a href="index.html">Movieapp/utils</a> movieListFixture.js</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>2/2</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>0/0</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>0/0</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>2/2</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a>
<a name='L33'></a><a href='#L33'>33</a>
<a name='L34'></a><a href='#L34'>34</a>
<a name='L35'></a><a href='#L35'>35</a>
<a name='L36'></a><a href='#L36'>36</a>
<a name='L37'></a><a href='#L37'>37</a>
<a name='L38'></a><a href='#L38'>38</a>
<a name='L39'></a><a href='#L39'>39</a>
<a name='L40'></a><a href='#L40'>40</a>
<a name='L41'></a><a href='#L41'>41</a>
<a name='L42'></a><a href='#L42'>42</a>
<a name='L43'></a><a href='#L43'>43</a>
<a name='L44'></a><a href='#L44'>44</a>
<a name='L45'></a><a href='#L45'>45</a>
<a name='L46'></a><a href='#L46'>46</a>
<a name='L47'></a><a href='#L47'>47</a>
<a name='L48'></a><a href='#L48'>48</a></td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { configureStore } from "@reduxjs/toolkit";
import movieRatingReducer from "../redux/movieSlicer"
&nbsp;
const movies = {
    425909: {
      title: "Ghostbusters: Afterlife",
      posterPath: "/sg4xJaufDiQl7caFEskBtQXfD4x.jpg",
      overview:
        "When a single mom and her two kids arrive in a small town, they begin to discover their connection to the original Ghostbusters and the secret legacy their grandfather left behind.",
    },
    438695: {
      title: "Sing 2",
      posterPath: "/aWeKITRFbbwY8txG5uCj4rMCfSP.jpg",
      overview:
        "Buster and his new cast now have their sights set on debuting a new show at the Crystal Tower Theater in glamorous Redshore City. But with no connections, he and his singers must sneak into the Crystal Entertainment offices, run by the ruthless wolf mogul Jimmy Crystal, where the gang pitches the ridiculous idea of casting the lion rock legend Clay Calloway in their show. Buster must embark on a quest to find the now-isolated Clay and persuade him to return to the stage.",
    },
    460458: {
      title: "Resident Evil: Welcome to Raccoon City",
      posterPath: "/7uRbWOXxpWDMtnsd2PF3clu65jc.jpg",
      overview:
        "Once the booming home of pharmaceutical giant Umbrella Corporation, Raccoon City is now a dying Midwestern town. The company’s exodus left the city a wasteland…with great evil brewing below the surface. When that evil is unleashed, the townspeople are forever…changed…and a small group of survivors must work together to uncover the truth behind Umbrella and make it through the night.",
    },
    512195: {
      title: "Red Notice",
      posterPath: "/lAXONuqg41NwUMuzMiFvicDET9Y.jpg",
      overview:
        "An Interpol-issued Red Notice is a global alert to hunt and capture the world's most wanted. But when a daring heist brings together the FBI's top profiler and two rival criminals, there's no telling what will happen.",
    },
  };
&nbsp;
  export const mockStore = configureStore({
    reducer: {
      movieRatings: movieRatingReducer,
    },
    preloadedState: {
      movieRatings: {
        sessionID: "",
        username: "",
        nmbMovies: 0,
        loading: false,
        movies: movies,
        ratings: {},
        totalResults: {},
        sortedIDs: [],
        participants: [],
      },
    },
  });</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Sun Jan 30 2022 18:21:24 GMT+0100 (sentraleuropeisk normaltid)
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    