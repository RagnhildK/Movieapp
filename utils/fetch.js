const baseURL =
  "https://api.themoviedb.org/3/movie/popular?api_key=72c828341c35299683ab545ba90e7f50&language=en-US&page=1";


export const getMovie = async (handleResponse) => {
  try {
    const response = await fetch(baseURL);
    const json = await response.json();
    handleResponse(json);
  } catch (error) {
    console.log(error);
  }
};
