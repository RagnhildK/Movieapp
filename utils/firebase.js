// Import the functions you need from the SDKs you need
import { initializeApp } from "firebase/app";
// import { getAnalytics } from "firebase/analytics";
import {
  getFirestore,
  doc,
  setDoc,
  collection,
  addDoc,
  where,
  query,
  getDoc,
  getDocs,
} from "firebase/firestore";

// TODO: Add SDKs for Firebase products that you want to use
// https://firebase.google.com/docs/web/setup#available-libraries

// Your web app's Firebase configuration
// For Firebase JS SDK v7.20.0 and later, measurementId is optional
const firebaseConfig = {
  apiKey: "AIzaSyCANKrut9VEYA6j9Erq9KEtCgekuNM8Cbs",
  authDomain: "movieapp-903cc.firebaseapp.com",
  projectId: "movieapp-903cc",
  storageBucket: "movieapp-903cc.appspot.com",
  messagingSenderId: "658249651052",
  appId: "1:658249651052:web:b2e8e233e3e95753025949",
  measurementId: "G-C4KJBS2XJY",
};

// Initialize Firebase
const app = initializeApp(firebaseConfig);
// const analytics = getAnalytics(app);

const db = getFirestore();

const UserColl = collection(db, "Users");
// const SessionColl = collection(db, "Participants");

export async function addUser(name) {
  const data = {
    username: name,
  };
  await addDoc(UserColl, data);
}

export async function checkIfUserExists(user) {
  const q = query(UserColl, where("username", "==", user));
  const a = await getDocs(q);
  return a.size != 0;
}

export async function addSession(name, sessionID, nmbMovies) {
  const data = {
    [name]: {},
  };
  const movies = {
    amount: nmbMovies,
  };
  await setDoc(doc(db, "Participants", sessionID), data);
  await setDoc(doc(db, "Movies", sessionID), movies);
}

export async function addParticipant(username, sessionID) {
  const data = {
    [username]: {},
  };
  await setDoc(doc(db, "Participants", sessionID), data, { merge: true });
}

export async function getNmbOfMovies(sessionID, handleResponse) {
  const a = doc(db, "Movies", sessionID);

  let movies;
  await getDoc(a).then((doc) => {
    movies = doc.data();
  });
  await handleResponse(movies);
}

export async function updateRatings(username, ratings, sessionID) {
  const data = {
    [username]: ratings,
  };
  await setDoc(doc(db, "Participants", sessionID), data, { merge: true });
}

export async function getRatings(sessionID, handleResponse) {
  const a = doc(db, "Participants", sessionID);
  let ratings;
  await getDoc(a).then((doc) => {
    ratings = doc.data();
  });
  await handleResponse(ratings);
}

// export async function checkIfUserExists() {
//   const docRef = doc(db, "Users", "");
//   const a = await getDoc();
//   console.log(a);
//   a.forEach((doc) => {
//     // doc.data() is never undefined for query doc snapshots
//     console.log(doc.id, " => ", doc.data());
//   });
// }
