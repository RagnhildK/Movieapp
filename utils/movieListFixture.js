import { configureStore } from "@reduxjs/toolkit";
import movieRatingReducer from "../redux/movieSlicer"

const movies = {
    425909: {
      title: "Ghostbusters: Afterlife",
      posterPath: "/sg4xJaufDiQl7caFEskBtQXfD4x.jpg",
      overview:
        "When a single mom and her two kids arrive in a small town, they begin to discover their connection to the original Ghostbusters and the secret legacy their grandfather left behind.",
    },
    438695: {
      title: "Sing 2",
      posterPath: "/aWeKITRFbbwY8txG5uCj4rMCfSP.jpg",
      overview:
        "Buster and his new cast now have their sights set on debuting a new show at the Crystal Tower Theater in glamorous Redshore City. But with no connections, he and his singers must sneak into the Crystal Entertainment offices, run by the ruthless wolf mogul Jimmy Crystal, where the gang pitches the ridiculous idea of casting the lion rock legend Clay Calloway in their show. Buster must embark on a quest to find the now-isolated Clay and persuade him to return to the stage.",
    },
    460458: {
      title: "Resident Evil: Welcome to Raccoon City",
      posterPath: "/7uRbWOXxpWDMtnsd2PF3clu65jc.jpg",
      overview:
        "Once the booming home of pharmaceutical giant Umbrella Corporation, Raccoon City is now a dying Midwestern town. The company’s exodus left the city a wasteland…with great evil brewing below the surface. When that evil is unleashed, the townspeople are forever…changed…and a small group of survivors must work together to uncover the truth behind Umbrella and make it through the night.",
    },
    512195: {
      title: "Red Notice",
      posterPath: "/lAXONuqg41NwUMuzMiFvicDET9Y.jpg",
      overview:
        "An Interpol-issued Red Notice is a global alert to hunt and capture the world's most wanted. But when a daring heist brings together the FBI's top profiler and two rival criminals, there's no telling what will happen.",
    },
  };

  export const mockStore = configureStore({
    reducer: {
      movieRatings: movieRatingReducer,
    },
    preloadedState: {
      movieRatings: {
        sessionID: "",
        username: "",
        nmbMovies: 0,
        loading: false,
        movies: movies,
        ratings: {},
        totalResults: {},
        sortedIDs: [],
        participants: [],
      },
    },
  });