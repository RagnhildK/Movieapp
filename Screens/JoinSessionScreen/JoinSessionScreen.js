import React, { useState } from "react";
import { View, StyleSheet, Pressable, Text } from "react-native";
import { useDispatch } from "react-redux";
import { setSessionID, setUsername, setLoading } from "../../redux/movieSlicer";
import { TextInput } from "react-native-paper";
import * as Colors from "../../styles/colors";
import { addParticipant } from "../../firebase";

export default function JoinSessionScreen({ navigation }) {
  const [owner, setOwner] = useState("");
  const [username, setLocalUsername] = useState("");
  const [userError, setUserError] = useState(false);
  const [sessionError, setSessionError] = useState(false);

  const dispatch = useDispatch();

  const enterSession = () => {
    const validUsername = username != "" ? true : false;
    const validSession = owner != "" ? true : false;
    if (!validUsername) {
      if (!validSession) {
        setUserError(true);
        setSessionError(true);
      } else {
        setUserError(true);
        setSessionError(false);
      }
    } else {
      if (!validSession) {
        setSessionError(true);
        setUserError(false);
      } else {
        dispatch(setUsername(username));
        dispatch(setSessionID(owner));
        dispatch(setLoading(true));
        addParticipant(username, owner);
        navigation.navigate("RatingScreen");
      }
    }
  };

  return (
    <View style={styles.container}>
      <TextInput
        mode="outlined"
        label="Username"
        error={userError}
        onChangeText={(val) => setLocalUsername(val)}
        value={username}
        placeholder="Enter a username for the session..."
      />
      <TextInput
        mode="outlined"
        label="Username of session creator"
        error={sessionError}
        onChangeText={(val) => setOwner(val)}
        value={owner}
        placeholder="username of session creator..."
      />
      <Pressable style={styles.button} onPress={() => enterSession()}>
        <Text style={styles.buttonText}> Start rating</Text>
      </Pressable>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    backgroundColor: Colors.BEIGE,
  },
  button: {
    backgroundColor: Colors.BROWN_RED,
    margin: 10,
    padding: 10,
    maxWidth: 300,
    borderRadius: 15,
  },
  buttonText: {
    color: Colors.WHITE,
    textAlign: "center",
  },
});
