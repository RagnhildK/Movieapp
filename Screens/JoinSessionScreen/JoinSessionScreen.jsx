import React, { useState } from "react";
import { View, StyleSheet, Pressable, Text, SafeAreaView } from "react-native";
import { useDispatch } from "react-redux";
import { setSessionID, setUsername, setLoading } from "../../redux/movieSlicer";
import { TextInput } from "react-native-paper";
import * as Colors from "../../styles/colors";
import { addParticipant } from "../../utils/firebase";
import { Headline } from "react-native-paper";

export default function JoinSessionScreen({ navigation }) {
  const [localSessionID, setLocalSessionID] = useState("");
  const [localUsername, setLocalUsername] = useState("");
  const [userError, setUserError] = useState(false);
  const [sessionError, setSessionError] = useState(false);

  const dispatch = useDispatch();

  const enterSession = () => {
    const validUsername = localUsername != "" ? true : false;
    const validSession = localSessionID != "" ? true : false;
    if (!validUsername) {
      if (!validSession) {
        setUserError(true);
        setSessionError(true);
      } else {
        setUserError(true);
        setSessionError(false);
      }
    } else {
      if (!validSession) {
        setSessionError(true);
        setUserError(false);
      } else {
        dispatch(setUsername(localUsername));
        dispatch(setSessionID(localSessionID));
        dispatch(setLoading(true));
        addParticipant(localUsername, localSessionID);
        navigation.navigate("RatingScreen");
      }
    }
  };

  return (
    <SafeAreaView style={styles.container}>
      <Headline style={styles.heading1}>What's your nickname?</Headline>
      <TextInput
        mode="outlined"
        label="Nickname"
        error={userError}
        onChangeText={(val) => setLocalUsername(val)}
        value={localUsername}
        placeholder="Enter a nickname for the session..."
      />
      <Headline style={styles.heading2}>What's the name of the party?</Headline>
      <TextInput
        mode="outlined"
        label="Name of movie party"
        error={sessionError}
        onChangeText={(val) => setLocalSessionID(val)}
        value={localSessionID}
        placeholder="Enter a name for the party..."
      />
      <Pressable
        id="enterJoinedSession"
        style={styles.button}
        onPress={() => enterSession()}
      >
        <Text style={styles.buttonText}> Join party</Text>
      </Pressable>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    backgroundColor: Colors.DARK_PURPLE,
    paddingLeft: 20,
    paddingRight: 20,
  },
  heading1: {
    color: Colors.ORANGE_LIGHT,
    fontSize: 24,
    margin: 30,
    textAlign: "center",
    fontStyle: "italic",
  },
  heading2: {
    color: Colors.ORANGE_LIGHT,
    fontSize: 24,
    margin: 30,
    marginTop: 40,
    textAlign: "center",
    fontStyle: "italic",
  },
  button: {
    backgroundColor: Colors.PURPLE,
    marginTop: 20,
    margin: 10,
    padding: 10,
    maxWidth: 150,
    borderRadius: 10,
    alignSelf:"flex-end"

  },
  buttonText: {
    color: Colors.WHITE,
    textAlign: "center",
  },
});
