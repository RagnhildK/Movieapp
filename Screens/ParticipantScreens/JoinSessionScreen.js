import React, { useState } from "react";
import { View, Button, StyleSheet } from "react-native";
import { useDispatch } from "react-redux";
import { setSessionID, setUsername, setLoading } from "../../redux/movieSlicer";
import { TextInput } from "react-native-paper";
import * as Colors from "../../styles/colors";
import { addParticipant } from "../../firebase";

export default function JoinSessionScreen({ navigation }) {
  const [owner, setowner] = useState("");
  const [username, setLocalUsername] = useState("");
  const [userError, setUserError] = useState(false);
  const [sessionError, setSessionError] = useState(false);

  const dispatch = useDispatch();

  const enterSession = () => {
    //TODO deal med Ã¥ lage sjekk for valid session
    const validUserName = true;
    const validSession = true;
    if (!validUserName) {
      if (!validSession) {
        setUserError(true);
        setSessionError(true);
      } else {
        setUserError(true);
      }
    } else {
      if (!validSession) {
        setSessionError(true);
      } else {
        dispatch(setUsername(username));
        dispatch(setSessionID(owner));
        dispatch(setLoading(true));
        addParticipant(username, owner);
        navigation.navigate("RatingScreen");
      }
    }
  };

  // // if (registerUser(username)) {
  // if (true) {
  //   dispatch(setUsername(username));
  //   // if (joinASession(username, owner){ TODO make the function return true or false
  //   if (true) {
  //     dispatch(setSessionID(username));
  //     dispatch(setLoading(true));
  //     navigation.navigate("RatingScreen");
  //   }
  //   else{
  //     // TODO: error message that there is no session for that username
  //   }
  // } else {
  //   // TODO: error message that username is not available
  // }

  return (
    <View style={styles.container}>
      <TextInput
        mode="outlined"
        label="Username"
        error={userError}
        onChangeText={(val) => setLocalUsername(val)}
        value={username}
        placeholder="Enter a username for the session..."
      />
      <TextInput
        mode="outlined"
        label="Username of session creator"
        error={sessionError}
        onChangeText={(val) => setInput(val)}
        value={owner}
        placeholder="username of session creator..."
      />
      <Button
        color={Colors.BROWN_RED}
        title="Start rating"
        onPress={() => enterSession()}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    backgroundColor: Colors.BEIGE,
  },
});
