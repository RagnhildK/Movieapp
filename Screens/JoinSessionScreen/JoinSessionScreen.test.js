import React from "react";
import renderer from "react-test-renderer";
import JoinSessionScreen from "./JoinSessionScreen";
import { Provider } from "react-redux";
import { mockStore } from ".././../utils/movieListFixture";

jest.useFakeTimers();

it("JoinSessionScreen snapshot", async () => {
  const tree = renderer
    .create(
      <Provider store={mockStore}>
        <JoinSessionScreen />
      </Provider>
    )
    .toJSON();
  await expect(tree).toMatchSnapshot();
});

// test("Press the button to start rating", () => {
//   const mockCallback = jest.fn((x) => 42 + x);
//   enterSession([0, 1], mockCallback);
// 
//   // The mock function is called twice
//   expect(mockCallback.mock.calls.length).toBe(2);
// });