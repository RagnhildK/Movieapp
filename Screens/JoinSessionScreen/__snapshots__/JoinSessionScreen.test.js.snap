// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoinSessionScreen snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F4F1EC",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={Object {}}
  >
    <View
      pointerEvents="none"
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 6,
          },
          Object {
            "backgroundColor": "#f6f6f6",
            "borderColor": "rgba(0, 0, 0, 0.54)",
            "borderRadius": 4,
            "borderWidth": 1,
          },
        ]
      }
      testID="text-input-outline"
    />
    <View>
      <View
        style={
          Array [
            Object {
              "paddingBottom": 0,
            },
            Object {
              "minHeight": 64,
              "paddingTop": 8,
            },
          ]
        }
      >
        <View
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
              "zIndex": 3,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "backgroundColor": "#f6f6f6",
                "bottom": 4,
                "left": 10,
                "maxHeight": 2,
                "opacity": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
                "width": 12,
              }
            }
          />
          <Text
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "#f6f6f6",
                "color": "transparent",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 18,
                "maxWidth": -28,
                "opacity": 0,
                "paddingHorizontal": 0,
                "position": "absolute",
                "textAlign": "left",
                "top": 37,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                  Object {
                    "scaleY": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            Username
          </Text>
          <Text
            numberOfLines={1}
            onLayout={[Function]}
            style={
              Object {
                "color": "#6200ee",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "opacity": 0,
                "paddingHorizontal": 14,
                "position": "absolute",
                "textAlign": "left",
                "top": 36,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            Username
          </Text>
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "opacity": 0,
                "paddingHorizontal": 14,
                "position": "absolute",
                "textAlign": "left",
                "top": 36,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            Username
          </Text>
        </View>
        <TextInput
          allowFontScaling={true}
          editable={true}
          multiline={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder=""
          placeholderTextColor="rgba(0, 0, 0, 0.54)"
          rejectResponderTermination={true}
          selectionColor="#6200ee"
          style={
            Array [
              Object {
                "flexGrow": 1,
                "margin": 0,
                "paddingHorizontal": 14,
                "zIndex": 1,
              },
              Object {
                "height": 56,
              },
              Object {
                "paddingBottom": 0,
                "paddingTop": 0,
              },
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              false,
              Array [
                Object {},
              ],
            ]
          }
          testID="text-input-outlined"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
  <View
    style={Object {}}
  >
    <View
      pointerEvents="none"
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 6,
          },
          Object {
            "backgroundColor": "#f6f6f6",
            "borderColor": "rgba(0, 0, 0, 0.54)",
            "borderRadius": 4,
            "borderWidth": 1,
          },
        ]
      }
      testID="text-input-outline"
    />
    <View>
      <View
        style={
          Array [
            Object {
              "paddingBottom": 0,
            },
            Object {
              "minHeight": 64,
              "paddingTop": 8,
            },
          ]
        }
      >
        <View
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
              "zIndex": 3,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "backgroundColor": "#f6f6f6",
                "bottom": 4,
                "left": 10,
                "maxHeight": 2,
                "opacity": 0,
                "position": "absolute",
                "right": 0,
                "top": 6,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
                "width": 12,
              }
            }
          />
          <Text
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "#f6f6f6",
                "color": "transparent",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 18,
                "maxWidth": -28,
                "opacity": 0,
                "paddingHorizontal": 0,
                "position": "absolute",
                "textAlign": "left",
                "top": 37,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                  Object {
                    "scaleY": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            Username of session creator
          </Text>
          <Text
            numberOfLines={1}
            onLayout={[Function]}
            style={
              Object {
                "color": "#6200ee",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "opacity": 0,
                "paddingHorizontal": 14,
                "position": "absolute",
                "textAlign": "left",
                "top": 36,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            Username of session creator
          </Text>
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.54)",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "left": 0,
                "opacity": 0,
                "paddingHorizontal": 14,
                "position": "absolute",
                "textAlign": "left",
                "top": 36,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 1,
                  },
                ],
                "writingDirection": "ltr",
              }
            }
          >
            Username of session creator
          </Text>
        </View>
        <TextInput
          allowFontScaling={true}
          editable={true}
          multiline={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder=""
          placeholderTextColor="rgba(0, 0, 0, 0.54)"
          rejectResponderTermination={true}
          selectionColor="#6200ee"
          style={
            Array [
              Object {
                "flexGrow": 1,
                "margin": 0,
                "paddingHorizontal": 14,
                "zIndex": 1,
              },
              Object {
                "height": 56,
              },
              Object {
                "paddingBottom": 0,
                "paddingTop": 0,
              },
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              false,
              Array [
                Object {},
              ],
            ]
          }
          testID="text-input-outlined"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#4E1C00",
        "borderRadius": 15,
        "margin": 10,
        "maxWidth": 300,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "textAlign": "center",
        }
      }
    >
       Start rating
    </Text>
  </View>
</View>
`;
